ios/RNSSearchBar.mm

```diff
@@ -163,26 +252,75 @@ - (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar
   [self resignFirstResponder];
   [self hideCancelButton];
 
-  if (self.onCancelButtonPress) {
-    self.onCancelButtonPress(@{});
+  [self emitOnCancelButtonPressEvent];
+  [self emitOnChangeTextEventWithText:_controller.searchBar.text];
+}
+#endif // !TARGET_OS_TV
+
+#pragma mark-- Fabric specific
+
+#ifdef RN_FABRIC_ENABLED
+- (void)updateProps:(facebook::react::Props::Shared const &)props
+           oldProps:(facebook::react::Props::Shared const &)oldProps
+{
+  const auto &oldScreenProps = *std::static_pointer_cast<const facebook::react::RNSSearchBarProps>(_props);
+  const auto &newScreenProps = *std::static_pointer_cast<const facebook::react::RNSSearchBarProps>(props);
+
+  [self setHideWhenScrolling:newScreenProps.hideWhenScrolling];
+
+  if (oldScreenProps.cancelButtonText != newScreenProps.cancelButtonText) {
+    [self setCancelButtonText:RCTNSStringFromStringNilIfEmpty(newScreenProps.cancelButtonText)];
   }
-  if (self.onChangeText) {
-    self.onChangeText(@{
-      @"text" : _controller.searchBar.text,
-    });
+
+  if (oldScreenProps.obscureBackground != newScreenProps.obscureBackground) {
+    [self setObscureBackground:newScreenProps.obscureBackground];
+  }
+
+  if (oldScreenProps.hideNavigationBar != newScreenProps.hideNavigationBar) {
+    [self setHideNavigationBar:newScreenProps.hideNavigationBar];
+  }
+
+  if (oldScreenProps.placeholder != newScreenProps.placeholder) {
+    [self setPlaceholder:RCTNSStringFromStringNilIfEmpty(newScreenProps.placeholder)];
+  }
+
+  if (oldScreenProps.autoCapitalize != newScreenProps.autoCapitalize) {
+    [self setAutoCapitalize:[RNSConvert UITextAutocapitalizationTypeFromCppEquivalent:newScreenProps.autoCapitalize]];
   }
+
+  // TODO: implement props with Color types
```

Comment:
Is there a problem with this? I think you just need to use `RCTUIColorFromSharedColor` function iirc

Reply:
I just did not know out of hand how colours work (they're passed as strings in JS, but I guess they're internally converted somewhere) and decided to postpone implementing these props (but I will add them in this PR). Thanks for telling me the right function.